packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -335,6 +335,16 @@ RestrictedApplication.propTypes = {
   DEV_ONLY__loadNavbarMenuConfig: PropTypes.func,
 };
 
+const shallowlyCoerceBooleanValues = obj =>
+  Object.keys(obj).reduce((updatedObj, key) => {
+    const value = obj[key];
+    const isBoolean = value === 'true' || value === 'false';
+    return {
+      ...updatedObj,
+      [key]: isBoolean ? value === 'true' : value,
+    };
+  }, {});
```

Comment:
I know this may be a bit over the top but I wanted to suggest it and at least give the full function's code for an easy copy. I just prefer very clear names for things 😄 

```js
const isBooleanAsString = environmentValue =>
  environmentValue === "true" || environmentValue === "false";

const shallowlyCoerceBooleanValues = defaultMemoize(
  uncoercedEnvironmentValues =>
    Object.keys(uncoercedEnvironmentValues).reduce(
      (uncoercedEnvironmentValues, keyOfUncoercedEnvironmentValue) => {
        const uncoercedEnvironmentValue =
          uncoercedEnvironmentValues[keyOfUncoercedEnvironmentValue];

        return {
          ...coercedEnvironmentValues,
          [key]: isBooleanAsString(uncoercedEnvironmentValue)
            ? uncoercedEnvironmentValue === "true"
            : uncoercedEnvironmentValue
        };
      },
      {}
    )
);
```